frontend/controller/service-worker.js

```diff
@@ -68,6 +68,10 @@ const waitUntilSwReady = () => {
     const messageChannel = new MessageChannel()
     messageChannel.port1.onmessage = (event) => {
       if (event.data.type === 'ready') {
+        // For backward- and forward-compatibility (`currentSyncs` may not be
+        // defined in a different SW version)
+        if (!event.data.currentSyncs) return
```

Comment:
Actually, one more question: should this call `reject` before calling `return`?

Reply:
No, but `resolve` should be called. Fixed.